src/renderers/webgl/WebGLCubeMaps.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @author mrdoob / http://mrdoob.com/
+ */
+
+import { CubeReflectionMapping, CubeRefractionMapping, EquirectangularReflectionMapping, EquirectangularRefractionMapping } from '../../constants.js';
+import { WebGLCubeRenderTarget } from "../WebGLCubeRenderTarget.js";
+
+function WebGLCubeMaps( renderer ) {
+
+	let cubemaps = new WeakMap();
+
+	function mapTextureMapping( texture, mapping ) {
+
+		if ( mapping === EquirectangularReflectionMapping ) {
+
+			texture.mapping = CubeReflectionMapping;
+
+		} else if ( mapping === EquirectangularRefractionMapping ) {
+
+			texture.mapping = CubeRefractionMapping;
+
+		}
+
+		return texture;
+
+	}
+
+	function get( texture ) {
+
+		if ( texture && texture.isTexture ) {
+
+			const mapping = texture.mapping;
+
+			if ( mapping === EquirectangularReflectionMapping || mapping === EquirectangularRefractionMapping ) {
+
+				if ( cubemaps.has( texture ) ) {
+
+					const cubemap = cubemaps.get( texture ).texture;
+					return mapTextureMapping( cubemap, texture.mapping );
+
+				} else {
+
+					const image = texture.image;
+
+					if ( texture.isDataTexture === true || image.complete === true ) {
```

Comment:
On last question: Do we have an example where the code uses a compressed equirectangular texture (e.g. DDS)?Right now, `CompressedTexture` is not supported by `WebGLCubeMaps.get()`. How about addding `isCompressedTexture` to the `if` statement?

Reply:
Yes, I guess that should be safe to add.
I'll add it. Thanks!

Reply:
Ended up changing the check to just `if ( image.height > 0 ) {`. I double checked that dds textures work.